pkg/cloudprovider/providers/photon/photon.go

```diff
@@ -0,0 +1,550 @@
+}
+
+func (pc *PCCloud) AddSSHKeyToAllInstances(user string, keyData []byte) error {
+	return errors.New("unimplemented")
+}
+
+func (pc *PCCloud) CurrentNodeName(hostname string) (k8stypes.NodeName, error) {
+	pc.localK8sHostname = hostname
+	return k8stypes.NodeName(hostname), nil
+}
+
+func getInstanceID(name string, projID string) (string, error) {
+	var vmID string
+	var err error
+
+	if overrideIP == true {
+		vmID, err = getVMIDbyIP(projID, name)
+	} else {
+		vmID, err = getVMIDbyNodename(projID, name)
+	}
+	if err != nil {
+		return "", err
+	}
+
+	if vmID == "" {
+		err = cloudprovider.InstanceNotFound
+	}
+
+	return vmID, err
+}
+
+// ExternalID returns the cloud provider ID of the specified instance (deprecated).
+func (pc *PCCloud) ExternalID(nodeName k8stypes.NodeName) (string, error) {
+	name := string(nodeName)
+	if name == pc.localK8sHostname {
+		return pc.localInstanceID, nil
+	} else {
+		ID, err := getInstanceID(name, pc.projID)
+		if err != nil {
+			return ID, logError("getInstanceID failed for ExternalID", err)
+		} else {
+			return ID, nil
+		}
+	}
+}
+
+// InstanceID returns the cloud provider ID of the specified instance.
+func (pc *PCCloud) InstanceID(nodeName k8stypes.NodeName) (string, error) {
+	name := string(nodeName)
+	if name == pc.localK8sHostname {
+		return pc.localInstanceID, nil
+	} else {
+		ID, err := getInstanceID(name, pc.projID)
+		if err != nil {
+			return ID, logError("getInstanceID failed for InstanceID", err)
+		} else {
+			return ID, nil
+		}
+	}
+}
+
+func (pc *PCCloud) InstanceType(nodeName k8stypes.NodeName) (string, error) {
+	return "", nil
+}
+
+func (pc *PCCloud) Clusters() (cloudprovider.Clusters, bool) {
+	return nil, true
+}
+
+// ProviderName returns the cloud provider ID.
+func (pc *PCCloud) ProviderName() string {
+	return ProviderName
+}
+
+// LoadBalancer returns an implementation of LoadBalancer for Photon Controller.
+func (pc *PCCloud) LoadBalancer() (cloudprovider.LoadBalancer, bool) {
+	return nil, false
+}
+
+// Zones returns an implementation of Zones for Photon Controller.
+func (pc *PCCloud) Zones() (cloudprovider.Zones, bool) {
+	glog.V(4).Info("Claiming to support Zones")
+	return pc, true
+}
+
+func (pc *PCCloud) GetZone() (cloudprovider.Zone, error) {
+	return pc.Zone, nil
+}
+
+// Routes returns a false since the interface is not supported for photon controller.
+func (pc *PCCloud) Routes() (cloudprovider.Routes, bool) {
+	return nil, false
+}
+
+// ScrubDNS filters DNS settings for pods.
+func (pc *PCCloud) ScrubDNS(nameservers, searches []string) (nsOut, srchOut []string) {
+	return nameservers, searches
+}
+
+// Attaches given virtual disk volume to the compute running kubelet.
+func (pc *PCCloud) AttachDisk(pdID string, nodeName k8stypes.NodeName) error {
+	name := string(nodeName)
+	operation := &photon.VmDiskOperation{
+		DiskID: pdID,
+	}
+
+	var vmID string
+	var err error
+	if name == pc.localK8sHostname {
```

Comment:
Instead of checking for localK8sHostname, you can use something like,
name = pc.InstanceID(name)